File: src/main/java/com/atsuishio/superbwarfare/init/ModAttachments.java
package com.atsuishio.superbwarfare.init;

import com.atsuishio.superbwarfare.Mod;
import com.atsuishio.superbwarfare.capability.player.PlayerVariable;
import net.fabricmc.fabric.api.attachment.v1.AttachmentRegistry;
import net.fabricmc.fabric.api.attachment.v1.AttachmentType;
import net.minecraft.nbt.CompoundTag;

public class ModAttachments {

    public static final AttachmentType<PlayerVariable> PLAYER_VARIABLE = AttachmentRegistry.create(
            Mod.loc("player_variable"),
            builder -> builder
                    .initializer(PlayerVariable::new)
                    .persistent(CompoundTag.CODEC.xmap(
                            tag -> {
                                var var = new PlayerVariable();
                                var.readFromNBT(tag);
                                return var;
                            },
                            PlayerVariable::writeToNBT
                    ))
                    .copyOnDeath()
    );

    public static void init() {
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/init/ModAttachments.kt
package com.atsuishio.superbwarfare.init

import com.atsuishio.superbwarfare.Mod
import com.atsuishio.superbwarfare.item.attachment.AttachmentItem
import net.minecraft.world.item.Item
import java.util.function.Supplier

object ModAttachments {
    @JvmField
    val OEM_STOCK_STANDARD: Item = register("oem_stock_standard")

    @JvmField
    val MAGAZINE_STANDARD: Item = register("magazine_standard")

    @JvmField
    val ITEMS: List<Item> = listOf(OEM_STOCK_STANDARD, MAGAZINE_STANDARD)

    private fun register(id: String): Item {
        return Registration.item(id, Supplier { AttachmentItem("${Mod.MODID}:$id") })
    }

    @JvmStatic
    fun init() = Unit
}
